tests/unit/etlng/LoadBalancerTests.cpp

```diff
@@ -144,14 +144,16 @@ struct LoadBalancerConstructorNgTests : util::prometheus::WithPrometheus, MockBa
     makeLoadBalancer()
     {
         auto const cfg = getParseLoadBalancerConfig(configJson_);
-        return std::make_unique<LoadBalancer>(
+        auto lb = std::make_unique<LoadBalancer>(
             cfg,
             ioContext_,
             backend_,
             subscriptionManager_,
             networkManager_,
             [this](auto&&... args) -> SourcePtr { return sourceFactory_(std::forward<decltype(args)>(args)...); }
         );
+        lb->setSeed(0);
```

Comment:
Maybe it would be better to create an interface for `RandomGenerator` and inject mock.

Reply:
I think it's out of scope of this PR and for now overcomplicates things.

Reply:
@kuznetsss I won't be able to use interface here, because `uniform` function is a template

Reply:
Added an interface another way around and moved this implementation to derived class.